n3fit/src/n3fit/model_trainer.py

```diff
@@ -549,8 +563,18 @@ def _generate_observables(
             pos_initial, pos_multiplier = _LM_initial_and_multiplier(
                 all_pos_initial, all_pos_multiplier, max_lambda, positivity_steps
             )
+            replica_masks = np.stack([pos_dict["trmask"] for i in range(len(self.exp_info))])
+            training_data = np.stack(
+                [pos_dict["expdata"].flatten() for i in range(len(self.exp_info))]
+            )
```

Comment:
Please avoid list comprehensions.

Reply:
Done in [df26706](https://github.com/NNPDF/nnpdf/pull/1788/commits/df267066a0742a1d89ff708dc3b6e4b0685b3fe6)